src/frontend/app/features/applications/deploy-application/deploy-application-step2/deploy-application-step2.component.ts

```diff
@@ -274,6 +281,31 @@ export class DeployApplicationStep2Component
     this.subscriptions.push(setInitialSourceType$.subscribe());
     this.subscriptions.push(setSourceTypeModel$.subscribe());
     this.subscriptions.push(setProjectName.subscribe());
+
+    this.subscriptions.push(this.sourceSelectionForm.valueChanges.subscribe(form => {
```

Comment:
You should be about to do something like 


`this.suggestedRepos$ = this.sourceSelectionForm.valueChanges.pipe(`
`	map(form => form.projectName),`
`	startWith(''),`
`	pairwise(),`
`	filter(([oldName, newName]) => oldName !=== newName),`
`	switchMap(([oldName, newName]) => this.updateSuggestedRepositories(newName))`
`)`

and avoid the need for the extra subscription and lastProjectName.